agent/command/git.go

```diff
@@ -48,7 +49,9 @@ type gitFetchProject struct {
 
 	Token string `plugin:"expand" mapstructure:"token"`
 
+	// ShallowClone sets CloneDepth to 100, and is kept for backwards compatibility.
 	ShallowClone bool `mapstructure:"shallow_clone"`
+	CloneDepth   int  `mapstructure:"clone_depth"`
```

Comment:
If `ShallowClone` is set to true, and `CloneDepth` is also set, it seems like `ShallowClone` will have no effect. It might be worth it to print out a warning if this happens saying that it's not necessary to set shallow clone. I'll leave this up to you though.

Reply:
Good idea 👍